tools/lib/src/device.dart

```diff
@@ -0,0 +1,369 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:file/file.dart';
+import 'package:flutter_plugin_tools/src/common/core.dart';
+import 'package:flutter_plugin_tools/src/common/package_looping_command.dart';
+
+import 'syncable_process_runner.dart';
+import 'tizen_sdk.dart';
+
+export 'package:flutter_plugin_tools/src/common/package_looping_command.dart'
+    show PackageResult, RunState;
+
+/// A reference to a Tizen device(either physical or emulator) that can run
+/// Flutter applications.
+///
+/// Physical devices are connected when they're connected to host PC either
+/// with a cable or wirelessly. Emulators are connected when they're launched on
+/// host PC.
+class Device {
+  Device._(
+    this.name,
+    this.profile, {
+    required TizenSdk tizenSdk,
+    SyncableProcessRunner processRunner = const SyncableProcessRunner(),
+  })  : _tizenSdk = tizenSdk,
+        _processRunner = processRunner;
+
+  /// Creates a new physical Tizen device reference.
+  ///
+  /// Valid characters for a device name are [A-Za-z0-9-_].
+  factory Device.physical(
+    String name,
+    Profile profile, {
+    required TizenSdk tizenSdk,
+    SyncableProcessRunner processRunner = const SyncableProcessRunner(),
+  }) {
+    final Device device = Device._(
+      name,
+      profile,
+      tizenSdk: tizenSdk,
+      processRunner: processRunner,
+    );
+    device._id = device._findId();
+    if (device._id == null) {
+      throw Exception('$name($profile)\'s id is null.'
+          'Physical device references must be connected to host PC.');
+    }
+    return device;
+  }
+
+  /// Creates a new reference to a Tizen emulator device.
+  ///
+  /// Valid characters for a device name are [A-Za-z0-9-_].
+  factory Device.emulator(
+    String name,
+    Profile profile, {
+    required TizenSdk tizenSdk,
+    SyncableProcessRunner processRunner = const SyncableProcessRunner(),
+  }) =>
+      EmulatorDevice(
+        name,
+        profile,
+        tizenSdk: tizenSdk,
+        processRunner: processRunner,
+      );
+
+  /// The name of the device.
+  final String name;
+
+  /// A profile of the device.
+  final Profile profile;
+
+  final TizenSdk _tizenSdk;
+
+  String? _id;
+
+  final SyncableProcessRunner _processRunner;
+
+  /// The unqiue identifier assigned to a connected device.
+  String? get id => _id;
+
+  /// Whether this device is an emulator.
+  bool get isEmulator => false;
+
+  /// Whether this device is connected to host PC.
+  bool get isConnected {
+    _id = _findId();
+    return _id != null;
+  }
```

Comment:
I assumed that id is dynamically assigned by the sdb server since I couldn't find this info other than `sdb devices`. I'll fix the code so that it makes more sense.

Reply:
I accidentally merged this PR but has this issue been resolved? If not, please make another PR. Sorry.